server/websocket.go

```diff
@@ -955,9 +958,60 @@ func (s *WebsocketServer) onNewBlockAsync(hash string, height uint32) {
 	glog.Info("broadcasting new block ", height, " ", hash, " to ", len(s.newBlockSubscriptions), " channels")
 }
 
+func (s *WebsocketServer) publishNewBlockTxsByAddr(block *bchain.Block) {
+	s.addressSubscriptionsLock.Lock()
+	publishNewBlockTxs := func() bool {
+		for _, as := range s.addressSubscriptions {
+			for _, details := range as {
+				if details.publishNewBlockTxs {
+					return true
+				}
+			}
+		}
+		return false
+	}()
+	s.addressSubscriptionsLock.Unlock()
+	if publishNewBlockTxs {
+		for _, tx := range block.Txs {
+			tx := tx
+			go func() {
+				var tokenTransfers bchain.TokenTransfers
+				if s.chainParser.GetChainType() == bchain.ChainEthereumType {
+					tokenTransfers, _ = s.chainParser.EthereumTypeGetTokenTransfersFromTx(&tx)
+				}
+				vins := make([]bchain.MempoolVin, len(tx.Vin))
+				for i, vin := range tx.Vin {
+					vins[i] = bchain.MempoolVin{Vin: vin}
+				}
+				subscribed := s.getNewTxSubscriptions(vins, tx.Vout, tokenTransfers)
+				if len(subscribed) > 0 {
+					if csd, ok := tx.CoinSpecificData.(bchain.EthereumSpecificData); ok {
+						receipt, err := s.chain.EthereumTypeGetTransactionReceipt(tx.Txid)
+						if err != nil {
+							glog.Error("GetTransactionFromMempoolTx error ", err, " for ", tx.Txid)
+							return
```

Comment:
I think there are copy/paste issues... In addition to that, I am a bit surprised that the receipt is not already in the CoinSpecificData as it must be fetched during block processing to get the token transfers.

Reply:
Good catch, fixed: 61bd16f7f12de217bc6b5800bec1b59c554dce57

The receipt contains the logs relevant for processing transfers so we are able to detect addresses without any extra rpc calls required, however, the constructed receipt from `ethrpc.GetBlock` is missing the gas/status/fee data which is required to be fetched from the node to return the full `api.Tx`